### DISCUSSION TITLE
Video cannot be downloaded. Error: Exit code: 1

### DISCUSSION THREAD

**Discussion by rnfnrnfn** [aclap-dev/video-downloadhelper#442] [Help and Q&A]
### Your question

Hi all

I've used VDH for months very well and suddenly I found out that it doesn't work for some contents

Details are as below;

1) Error code
Exit code: 1
Error: Exit code: 1

    at as.receive (chrome-extension://pfoiagbblcbmognbkekfpodpidedkmcc/background/main.js:1:13157)
    at chrome-extension://pfoiagbblcbmognbkekfpodpidedkmcc/background/main.js:1:42622

2) Version and license detail

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: ko
license: accepted
key (only 16 first characters): aff3a30964c44d70
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\ILuvU",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

3) Video information

URL : https://b01-kr-naver-vod.pstatic.net/navertv/a/read/v2/VOD_ALPHA/navertv_2019_08_20_302/hls/b4c6b968-c316-11e9-ba0d-246e963a49b9.m3u8?_lsu_sa_=6bd5effc913d63d698d245f863c537ba1e7c3f388c08ff2f3d0787c0475f33559f227a2a67651701703e3d52d9373b8bb3a3edc7dd99dac83abc9613e5d0cb15d6177138c82121069fe12132ddc6822900e74d330d36ec314263eddbcdc5756d2c79da5119486e1d4bd71464e8a70136a128cd263a726bcc6b37c0a131d01c0c68c4d42256d90a7e91670d3bc21af2e5

4) Video Details

<html>
<body>
<header class="weh-header" style="box-sizing: border-box; display: block; min-height: 42px; line-height: 42px; background-repeat: no-repeat; background-position: 6px 6px; background-size: 30px; flex: 0 0 auto; background-color: rgb(248, 248, 248); color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; background-image: url(&quot;images/icon-128.png&quot;);"><span class="weh-header-title" style="box-sizing: border-box; padding-left: 42px; font-weight: 700; font-size: 1.2em;">개체 상세 정보</span><span class="weh-header-close" style="box-sizing: border-box; font-weight: 100; font-size: 1.5em; line-height: 1.5em; cursor: pointer; color: gray; padding: 0px 12px 0px 0px; float: right;">⊗</span></header><main style="box-sizing: border-box; display: block; flex: 1 1 auto; position: relative; overflow-y: auto; height: 974px; color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">
chunked | hls
-- | --
core_media | {"builder":"NakedHls""protocol":"hls""duration":83.817155"container":{"name":"Mp4""extension":"mp4""audio_only_extension":"mp3""defacto_codecs":{...}"supported_video_codecs":[...]"supported_audio_codecs":[...]"mimetype":{}}"av":{"audio":false"video":{"codec":{"name":"H264""type":"video""mimetype":{}"defacto_container":"Mp4"}"fps":{}"dimensions":{}"quality":{}"bitrate":{}}}}
created | 1714414671769
descrPrefix | HLS streaming
duration | 83.817155
extension | mp4
group | grp-459b47d0
headers | [0:{"name":"sec-ch-ua""value":""Chromium";v="124", "Google Chrome";v="124", "Not-A.Brand";v="99..."}1:{"name":"sec-ch-ua-mobile""value":"?0"}2:{"name":"User-Agent""value":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KH..."}3:{"name":"sec-ch-ua-platform""value":""Windows""}4:{"name":"Accept""value":"*/*"}5:{"name":"Origin""value":"https://search.naver.com"}6:{"name":"Sec-Fetch-Site""value":"cross-site"}7:{"name":"Sec-Fetch-Mode""value":"cors"}8:{"name":"Sec-Fetch-Dest""value":"empty"}9:{"name":"Referer""value":"https://search.naver.com/search.naver?ssc=tab.video.all&where=vi..."}10:{"name":"Accept-Encoding""value":"gzip, deflate, br, zstd"}11:{"name":"Accept-Language""value":"ko-KR,ko;q=0.9,en-US;q=0.8,en;q=0.7"}]
id | nakedhls:459b47d0
isPrivate | false
mediaManifest | https://b01-kr-naver-vod.pstatic.net/navertv/a/read/v2/VOD_ALPHA/navertv_2019_08_20_302/hls/b4c6b968-c316-11e9-ba0d-246e963a49b9.m3u8?_lsu_sa_=6a85e9fb61f66646d9daa5a56d0518bc7ea73008110fbf7035379ac8d70538e501248a9c6b35210590cb32d251315bef49116d751fe4523878851c80b11b7998919c19bee493fb0950cb0b69e8ca055299796e8b8e3ad8217edcbefe59c7f0225d823427549c30a0ebca8d5ae36e8877d4457fb51fc00958307ef69c20d409da6696839ba7c88d01f8d9e9d038c24117
originalId | network-probe:459b47d0
pageUrl | https://search.naver.com/search.naver?ssc=tab.video.all&where=video&sm=tab_jum&query=%ED%81%AC%EB%AA%BD
referrer | https://search.naver.com/search.naver?ssc=tab.video.all&where=video&sm=tab_jum&query=%ED%81%AC%EB%AA%BD
status | active
tabId | 701130501
thumbnailUrl | https://ssl.pstatic.net/sstatic/search/common/og_v3.png
title | 크몽 : 네이버 동영상검색
topUrl | https://search.naver.com/search.naver?ssc=tab.video.all&where=video&sm=tab_jum&query=%ED%81%AC%EB%AA%BD

</main>
</body>
</html>

I check my anti-virus app and firewall (inbound and outbound) already.
also beta version doesn't work for me.
is there anything that I can check more?

Thank you in advance.

### Addon & CoApp version and details

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: ko
license: accepted
key (only 16 first characters): aff3a30964c44d70
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\ILuvU",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3447, Firefox 125.0.3 64-bit, licensed VDH 8.2.2.8a1 beta, CoApp 2.0.19.  I do not work for VDH.  I am just another user.

Often, if you post the URL of a manifest, which is what you did, other people can use it. But there is a warning.  The long strings of gibberish often relate the URL to your specific visit to that web site.  In addition, the URLs expire.  I just tried visiting your manifest URL & got no result whatsoever, just a blank browser screen.  So, we need a different URL.  We need the URL of the web page on which the video is hosted, not the manifest URL fished out of the Hit Details.

If you want to get the latest beta, & I strongly recommend you do, read this: #391.  By coincidence, that thread also describes in detail how you can use a manifest URL to get a video . . .  SOMETIMES.  Like I said, in your particular case, your URL didn't work for me.  In that other discussion it happened to work.  The URL in that other discussion actually does still work, but it doesn't have all those gibberish strings attached.  Every web site does things differently.  There's no predicting.  In your case here, we need the URL of the web page where the video resides.

**Reply by rnfnrnfn** [reply]
thank you for your feedback.

here is the link for the page : https://tv.naver.com/v/10010315

**Reply by GrampaWildWilly** [reply]
Looks like the Korean version of YouTube.  Anyway, I also got an error trying to download that.  I also opened the Network Monitor to see what I might find.  It turned out there was an HLS master manifest in there.  That was a welcome surprise since YouTube doesn't give something so direct.  Normally, a master manifest lets you download stuff without any trouble.  But I ran ffprobe against the manifest & I got the error 403 Forbidden.  I've encountered this on occasion.  It's kind of a stealth implementation of DRM.  In order to read the manifest, you need a password & probably a user ID.  It seems the web page sends the relevant identification when you play the video in the web page, something I did successfully, although I let it play for only a few seconds, since I don't speak Korean.  But "outsiders" like us have no idea what sort of authentication to provide.  I suppose VDH is equally at a loss.  If you can get VDH to download some content from this site but not other content, that's just the way they organize their web site.  They're generous with some content & jerks about other content.  Oh well.  Looks like we're just out of luck.

Unless . . . @paulrouget Paul has other ideas.

**Reply by rnfnrnfn** [reply]
Thank you very much for your reply.

I appreciate the detailed explanation.

I'm glad to know that I'm not the only one who can't download and I will try another way.

I tried downloading from other video platforms and found that I can't even download videos from Facebook or Instagram.

https://www.facebook.com/watch/?v=1855608441547559

For example, I couldn't download the video above?

Or any other video on Instragram.
https://www.instagram.com/reels/C5IrwVLStxR/

(FYI, these are randomly selected videos)

I was wondering if you have any advice for these cases as well.

Thank you very much for your kind reply and attention. I hope you have a nice day.

**Reply by GrampaWildWilly** [reply]
I am not a member of any social network.  You'll have to get answers to these questions from somebody else.

**Comment by fxzfz** [comment]
I have also encountered this "Exit code: 1" issue starting from April. Maybe the developer really needs to give an update to this app.
moz-extension://88cba37b-8cb7-4aa9-a2f5-8f62998a0ab5/background/main.js:1
receive@moz-extension://88cba37b-8cb7-4aa9-a2f5-8f62998a0ab5/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://88cba37b-8cb7-4aa9-a2f5-8f62998a0ab5/background/main.js:1:42624

**Reply by tevslin** [reply]
I just installed in firefox and got this same error. II have installed the helper app. the website I'm trying to download from is https://reflect-video-ondemand-cpsd.cablecast.tv/CablecastPublicSite/show/10284?site=2

**Reply by GrampaWildWilly** [reply]
Guys.  Guys.  Did you read what's above?  People don't post here for other people to ignore it.  You're supposed to READ what other people post.  It's the whole point of a forum like this.  People post their experience for the benefit of others coming along later.  But if the others don't bother reading what's already here, that is the equivalent of wasting what's already here.  It's also kind of insulting to the people who take the time & put in the effort to offer help to fellow users.  Think about it.

I listed my operating environment in a post upthread here.  I'm not going to list it all again.  I'm going to make you read it.  Go find it & read it.  Operating system (with version), browser (with version), VDH (with version), CoApp (with version).  Did I mention that you should specify which versions you are using of the various pieces of the software?  You need to tell people what software you are running, including versions.  It takes barely 1 line.  It's not that hard to do.  It must be within your capabilities.

Then you need to tell us at least one video you're having problems with.  At least one of you did that.  I visited the short school committee meeting page & in my operating environment (I say again, go read what I wrote), it downloads just fine.

So you see, you could have saved yourself (not to mention me) a fair bit of time if you had only read what was already here.  If you want things to work, you need to be running something approximating what I am running.  And what exactly am I running?  Do I have to tell you again?  Go find it & read it.

**Reply by tevslin** [reply]
I am running Firefox 125.0.3(64bit) on Windows 11 
Device name	BOOK-P8T9TP4CS7
Processor	12th Gen Intel(R) Core(TM) i7-1260P   2.10 GHz
Installed RAM	16.0 GB (15.6 GB usable)
Device ID	918629C0-BC2C-4BF4-9DE0-9A75D2A7AE2B
Product ID	00330-80853-66620-AA702
System type	64-bit operating system, x64-based processor
Pen and touch	Pen and touch support with 10 touch points

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\tevsl",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

Althoug I was able (obce) to download the video I initially pointed to, I am totally unable to download video from this window although the video is correctly located https://reflect-video-ondemand-cpsd.cablecast.tv/cablecastapi/embed?show_id=10231

The reported problem is always exit code 1 moz-extension://a4d3f482-5b30-4b2c-b0ca-e1c4d6ace49b/background/main.js:1
receive@moz-extension://a4d3f482-5b30-4b2c-b0ca-e1c4d6ace49b/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://a4d3f482-5b30-4b2c-b0ca-e1c4d6ace49b/background/main.js:1:42626\

Thank you

**Reply by GrampaWildWilly** [reply]
Now we're getting somewhere.  You've included information about your hardware configuration.  That's usually not useful for solving these software problems.  Interesting, but not usually necessary.  But the rest is important to know.  Especially the URL.

I was able to get VDH 8.2.2.8a1 beta to download that video of over 3.5 hours in a bit over 4 minutes.  It played fine in VLC.  I didn't sit & watch the whole 3.5+ hours of it, just sampled it at intervals to make sure it was all there.  I let the last 10 seconds play to make sure it was all there, right to the final adjournment gavel.

But you are still not running the same VDH as I am.  You're running a version that doesn't work.  I'm running one that does.  You need to read my posts upthread here more carefully.  I explain in detail how to get the same version I'm running.  That's what you need to do.

**Reply by GrampaWildWilly** [reply]
Some slightly off-topic comments.  You are running W11 in 16G of RAM.  How does that perform?  Everything I've read leads me to believe you're pushing your luck with less than 32G.  But if it is good enough for your needs, I'm not going to criticize.

I also notice you do not have a VDH license.  I say you need to be prepared for this 3.5+ hour video to take something approximating that long & perhaps longer, a lot longer, to write the watermark into it.  I would expect your download to last the approximately 4 minutes it took me, but your watermark . . .

**Comment by tevslin** [comment]
W11 OK in 16G altho always looking for an excuse to upgrade. Probably wait and get machine with a GPU.

 

Don’t really care about the watermark because I only want the audio. I know I can’t get that directly w/out a VDH license but can see the url with VDH and download audio directly with ffmpeg. Probably will get the license though.

 

Thanks for looking at this,

**Reply by GrampaWildWilly** [reply]
You do know there is an audio-only download function in VDH now, right?  It's on the secondary menu.  Hover your mouse over a variant in the VDH menu to make an arrow appear on the right.  Then click the arrow to open the secondary menu.  There's 2 audio-only functions, one "quick" one regular.  But maybe those options aren't available if you don't have a license, which would surprise me.  In any case, it's kind of hard to put a watermark in an audio-only file so my comment about that would probably not apply.  I'd be curious to hear how that works for you, especially if you have finally installed the 8.2.2.8a1 beta.
